zathura/zathura.c

```diff
@@ -546,6 +546,49 @@ document_info_open(gpointer data)
   return FALSE;
 }
 
+const char*
+get_formatted_filename(zathura_t* zathura, const char* file_path, bool statusbar)
+{
+  bool basename_only = false;
+  if (statusbar) {
+    girara_setting_get(zathura->ui.session, "window-title-basename", &basename_only);
+  } else {
+    girara_setting_get(zathura->ui.session, "statusbar-basename", &basename_only);
+  }
+
+  if (basename_only == false) {
+
+    bool home_tilde = false;
+    if (statusbar) {
+      girara_setting_get(zathura->ui.session, "statusbar-home-tilde", &home_tilde);
+    } else {
+      girara_setting_get(zathura->ui.session, "window-title-home-tilde", &home_tilde);
+    }
+
+    if (home_tilde) {
+      char *home = getenv("HOME");
+      size_t home_len = home ? strlen(home) : 0;
+      size_t file_path_len = file_path ? strlen(file_path) : 0;
+
+      if (home_len > 1 && strncmp(home, file_path, home_len) == 0 && (!file_path[home_len] || file_path[home_len] == '/')) {
+        // Length should be total length of path - length of $HOME + 1 for '~' + 1 for '\0'
+        int tlen = file_path_len - home_len + 2;
+        char *tdir = g_try_malloc(sizeof(char) * tlen);
```

Comment:
Can you explain why this might cause a memory leak? Sorry, I'm not very experienced in C.